tests/phpunit/Env/EnvBackup.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Env;
+
+use function array_key_exists;
+use function getenv;
+use function Safe\putenv;
+use function Safe\sprintf;
+use Webmozart\Assert\Assert;
+
+final class EnvBackup
```

Comment:
On one hand you argue that we do not need an extra package, but now we reinvent what is already installed.

Why not use [`sebastianbergmann/global-state`](https://github.com/sebastianbergmann/global-state)?

Reply:
Because it is way more complex since is handling a lot more cases, and also is probably what, somehow, causes the issue in your PR. So I'm fine with declaring a tiny class to avoid that dependency, there is nothing to complex there

Reply:
Why not write our own test framework then, since we do not need all of the assertions in `phpunit/phpunit`?

Reply:
I'm a bit concerned because a lot of code has been added recently that has nothing to do with the primary concern of `infection/infection`.

Reply:
>Why not write our own test framework then, since we do not need all of the assertions in phpunit/phpunit?

Sorry how is that even remotely relevant? For starters we are heavily using PHPUnit and its features and nothing warrants to use another test framework.

Also for the records: it's not like I'm adding this because I don't like PHPUnit, but because we cannot make use of the globals back-up and because you and I could not figure out why and I don't want to spend 3 hours looking for it when a 50LoC class does the job.